blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/FavoriteImpl.java

```diff
@@ -16,27 +16,30 @@
 
     private final Object item;
     private final Link self;
+    private final LinkResolver linkResolver;
 
     public FavoriteImpl(Item item, Reachable parent) {
         this.self = parent.getLink().rel(Util.rawEncode(item.getFullName()));
+        this.linkResolver = Jenkins.getActiveInstance().getInjector().getInstance(LinkResolver.class);
 
+        final Link link = linkResolver.resolve(item);
 
-        // TODO: MBP nested inside folder won't work nor would a favorited folder is going to work
-        //TODO: Needs https://issues.jenkins-ci.org/browse/JENKINS-36286 to be fixed,
-        //      it should use LinkResolver for it to work in all cases
-        Object obj = null;
-        if(item instanceof WorkflowJob){
-            if(item.getParent() instanceof MultiBranchProject){
-                Link s = OrganizationImpl.INSTANCE.getLink().rel(String.format("pipelines/%s/branches/",
-                    ((MultiBranchProject) item.getParent()).getName()));
-                obj = new BranchImpl((Job) item, s);
+        BluePipeline pipeline = null;
+        for(BluePipelineFactory factory:BluePipelineFactory.all()){
```

Comment:
Could this loop be moved up somehow to a utility method? Seems it would be a common pattern to find a pipeline based on an `item`